### PR TITLE
Refactor/read binary parameter type should be asset load context 

### PR DESCRIPTION
**Pull request by seijiwjs**

### CODE REVIEW

File: src/SceneTree/AssetLoadContext.ts [hunk 1]
```diff
@@ -25,8 +25,12 @@ export class AssetLoadContext extends EventEmitter {
   protected postLoadCallbacks: Array<() => void>
   protected asyncCount: number
 
+
   addGeomToLayer: (geomItem: GeomItem, layer: string) => void
 
+
+  materialLibrary: any //TODO: added in AssetLoadContext refactor
+  lod: number
```
Comment (phtaylor) [inline src/SceneTree/AssetLoadContext.ts:33]: I don't think we actually need these 2 members. I think we can eliminate them. (I thought we had already removed lod)

Comment (seijiwjs) [reply src/SceneTree/AssetLoadContext.ts:34]: it looks like material library is used in materialFloatParam, and Material colorParam.

```
  readBinary(reader: BinReader, context: AssetLoadContext): void {
    super.readBinary(reader, context)

    const textureName = reader.loadStr()
    if (textureName != '') {
      console.log('Load Texture')
      this.setImage(context.materialLibrary.getImage(textureName))
    }
  }
```

Comment (phtaylor) [reply src/SceneTree/AssetLoadContext.ts:34]: I think the asset load context already has a reference to the asset itself. (I think)
so with that, we can access the material library (.getMaterialLibrary())

Comment (phtaylor) [reply src/SceneTree/AssetLoadContext.ts:34]: lets remove all reference to 'lod' from the material library. We are not actually using it anymore. No one will notice its gone.

**Review by phtaylor — APPROVED** [review]

**Comment by phtaylor** [comment]
great work

**Pull request merged**